back/src/app/controllers/user.ts

```diff
@@ -201,4 +184,91 @@ export default class userController {
 	static getUser(req: Request, res: Response) {
 		res.send(req.user);
 	}
+
+	static async create(req: Request, res: Response) {
+		let copyReq: any = req;
+		const user = new User();
+		user.email = req.body.email;
+		user.login = req.body.login;
+		user.firstName = req.body.firstName;
+		user.lastName = req.body.lastName;
+		user.password = req.body.password;
+		user.imageUrl = user.email;//**********************************
+		user.oauth = false;
+		if (user.isValid() && !(await user.isEmailTaken()) && copyReq['session'].validPic == true) {
+			await user.setPassword(user.password);
+			user.save();
+			res.status(201).send(true);
+			userController.movePicToUserPic(user.email, copyReq);
+			return;
+		}
+		else{
+			res.status(400).send("false");
+		}
+	}
+
+	static movePicToUserPic(email: any, copyReq: any){
+		fs.rename('/back/public/tmpValid/' + copyReq.session.random, '/back/public/userPic/' + email, (callback: any) =>{
+		});
+	}
+
+	static movePicToTmpValid(copyReq: any){
+		copyReq.files.file_jerome.mv('/back/public/tmpValid/'+ copyReq.session.random);
+	}
+
+	static getCookie(req: Request, res: Response){
+		let copyReq: any = req;
+		copyReq['session'].id  = true;
+		res.send("Your cookies sir!");
+	}
+
+	static validPicture(files: any){
+		let check: any = {};
+		check.size = false;
+		check.type = false;
+
+		if (files.file_jerome.mimetype == 'image/png' || files.file_jerome.mimetype == 'image/jpeg')
+			check.type = true;
+		else
+			check.type = false;
+
+		if (files.file_jerome.size <= 9999999)
+			check.size = true;
+		else
+			check.size = false;
+
+
+		if (check.size && check.type){
+			return true;
+		}
+		else {
+			return check;
+		}
+	}
+
+	static saveProfilePic(req: Request, res: Response){
+		let copyReq: any = req;
+		if (copyReq.files == null){
+			res.send("vide");
+		}
+		else{
+			let isValidPic: any = userController.validPicture(copyReq.files);
+			if (isValidPic === true){
+				copyReq['session'].validPic = true;
+				let random:any = Math.floor((Math.random() * 10000) + 1);
+				let maRep :any = {};
+				maRep.expressSig = random;
+				copyReq.session.random = random;
+				userController.movePicToTmpValid(copyReq);
+				maRep.status = "sucess";
+				res.send(maRep);
+			}
+			else {
+				copyReq['session'].validPic = false;
```

Comment:
![Screen Shot 2020-02-27 at 2 32 04 PM](https://user-images.githubusercontent.com/38002047/75450111-83290680-596e-11ea-9ae7-c530588b75bf.png)
wrong status , 200 is not for error, 401 in this case

Reply:
Done